### PR TITLE
MC-37807: Link to a file of a Product with Customizable Option(File) is not available throughout a multishipping checkout process

### LINKED ISSUE #31095: [Issue] MC-37807: Link to a file of a Product with Customizable Option(File) is not available throughout a multishipping checkout process
This issue is automatically created based on existing pull request: magento/magento2#31066: MC-37807: Link to a file of a Product with Customizable Option(File) is not available throughout a multishipping checkout process

### Preconditions (*)
1. Magento 2.4-develop

### Steps to reproduce (*)
1. Go to the Storefront as the created Customer
2. Add the created Product to the Shopping Cart (Qty = 2), uploading a valid file
3. Open the mini-cart
4. Click the “View and Edit Cart” link
5. Click the “Check Out with Multiple Addresses” link
6. Select shipping addresses on the “Ship to Multiple Addresses”
7. Click the “Go to Shipping Information” button
8. Select Flat Rate - Fixed Shipping Method for the each address
9. Click the “Continue to Billing Information” button
10. Select “Check / Money order” Payment Method
11. Click the “Go to Review Your Order” button

### Expected result
There should be a link, enabling file downloading, throughout entire multishipping checkout process (name of the file)

### Actual result
There is no link enabling file downloading throughout entire multishipping checkout process (name of the file)

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#

### Manual testing scenarios (*)

Steps to Reproduce:

1. Go to the Storefront as the created Customer
2. Add the created Product to the Shopping Cart (Qty = 2), uploading a valid file
3. Open the mini-cart
4. Click the “View and Edit Cart” link
5. Click the “Check Out with Multiple Addresses” link
6. Select shipping addresses on the “Ship to Multiple Addresses”
7. Click the “Go to Shipping Information” button
8. Select Flat Rate - Fixed Shipping Method for the each address
9. Click the “Continue to Billing Information” button
10. Select “Check / Money order” Payment Method
11. Click the “Go to Review Your Order” button

**Expected results:**
There should be a link, enabling file downloading, throughout entire multishipping checkout process (name of the file)

 **Actual results:**
There is no link enabling file downloading throughout entire multishipping checkout process (name of the file)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#31095: MC-37807: Link to a file of a Product with Customizable Option(File) is not available throughout a multishipping checkout process

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all tests

File: dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php [hunk 1]
```diff
@@ -6,28 +6,55 @@
...
+ * Verify default items template
  */
-class OverviewTest extends \PHPUnit\Framework\TestCase
+class OverviewTest extends TestCase
 {
     /**
-     * @var \Magento\Multishipping\Block\Checkout\Overview
+     * @var Overview
+     */
+    protected $block;
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php]: Do you really need this be protected? Make it private if possible

File: dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php [hunk 2]
```diff
@@ -6,28 +6,55 @@
...
+
+    /**
+     * @var ObjectManagerInterface
      */
-    protected $_block;
+    protected $objectManager;
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php]: Do you really need this be protected? Make it private if possible

File: dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php [hunk 3]
```diff
@@ -6,28 +6,55 @@
...
+     * @var Product
+     */
+    private $product;
+
+    /**
+     * @var Item
+     */
+    private $item;
 
     protected function setUp(): void
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php:54]: Add @inheritDoc annotation

File: dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php [hunk 4]
```diff
@@ -37,33 +64,49 @@ protected function setUp(): void
...
+        $this->block->addChild('renderer.list', RendererList::class);
+        $this->block->getChildBlock(
             'renderer.list'
         )->addChild(
             'default',
-            \Magento\Checkout\Block\Cart\Item\Renderer::class,
+            Renderer::class,
             ['template' => 'cart/item/default.phtml']
         );
+        $this->quote = $this->objectManager->get(Quote::class);
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php]: Please use  ...Factory to create objects.
Same for next 2 lines

File: dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php [hunk 5]
```diff
@@ -37,33 +64,49 @@ protected function setUp(): void
...
-        /** @var $item \Magento\Quote\Model\Quote\Item */
-        $item = $this->_objectManager->create(\Magento\Quote\Model\Quote\Item::class);
-        /** @var $product \Magento\Catalog\Model\Product */
-        $product = $this->_objectManager->create(\Magento\Catalog\Model\Product::class);
-        $product->load(1);
-        $item->setProduct($product);
-        /** @var $quote \Magento\Quote\Model\Quote */
-        $quote = $this->_objectManager->create(\Magento\Quote\Model\Quote::class);
-        $item->setQuote($quote);
+        $product = $this->product->load('1');
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Multishipping/Block/Checkout/OverviewTest.php]: Have to use ProductRepository to get product instance by ID

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by engcom-Kilo — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

### Manual testing scenario
1. Create a Simple Product with **Customizable Option** type - **File**
cust_opt_type_file
2. Go to Storefront and Sign In as Customer
3. Open the previously crated products page and Upload a valid file
valid_file_upload
4. Add the previously created product to the Cart in a **Quantity of 2 items**
5. Click on mini cart and click **View and Edit Cart**
6. Click **Check Out with Multiple Addresses**
7.  Select different addresses in **Ship to Multiple Addresses** window
1_Ship to multiple
_Note that the link to the image files are valid_
8. Click **Go to Shipping Information**
9. Observe the links to the file in **Select Shipping Method**
2_sel_ship_meth
_Note that the link to the image files are valid_
10. Click **Continue to Billing Information**
11. Click **Go to Review Your Order**
12. Observe the links to the files in **Review Order** window
3_place_order
_Note that the links are valid and redirect to downloading the previously uploaded image_

**BEFORE** applying changes provided in this PR, only text with image name was displayed 
before

**AFTER** switching to the PR, the text is a link that redirects to file downloading. The links are valid through the whole multi shipping process.
The order could be Placed successfully.

**Comment by engcom-Bravo** [comment]
@magento create issue

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31066`.

**Pull request merged**